### PR TITLE
Docs: Add an FAQ section

### LINKED ISSUE #2160: Documentation doesn't include a FAQ section
There are a number of issues that our users ask us about in a periodic basis, for example:
 * Why can't I implement `Arbitrary` for an external type?
 * Kani produces confusing results after `kani::assume(false)`

We should have a FAQ section that answers these common questions, even if they could be answered by looking more closely to our documentation.

### PR DESCRIPTION
**Pull request by adpaco-aws**
### Description of changes: 

Adds an FAQ section to our documentation. This section appears after a separator and not as one of chapters.
In addition, this PR includes a couple of questions which I think will be frequent.

All questions appear at the beginning so they aren't difficult to find. Each question links to an answer that appears in the next section. **Another option is to have answers written in collapsible items** (similar to how we hide answers for some questions in the tutorial, which show a *Details* clickable element with an arrow). That would make it look cleaner but wouldn't allow us to link to a particular answer. Also, I've avoided bullet points for answers because that'd make it more difficult to write examples.

In other words, I'm not super-fond of the current layout - suggestions are very welcome.

### Resolved issues:

Resolves #2160 

### Related RFC:

Optional #ISSUE-NUMBER.

### Call-outs:

Call-outs done in description.

### Testing:

* How is this change tested? N/A

* Is this a refactor change? N/A

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [ ] Methods or procedures are documented
- [ ] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: docs/src/faq.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+**Question:** I implemented the `kani::Arbitrary` trait for a type that's not from my crate, and got the error
+`only traits defined in the current crate can be implemented for types defined outside of the crate`.
+What does this mean? What can I do? [Answer](#arbitrary-implementations-for-foreign-types)
+
+## Answers
+
+### Assuming a false condition
+
+`kani::assume(false)` (or `kani::assume(cond)` where `cond` is condition that results in `false` in the context of the program), won't cause errors in Kani.
```
Comment (zhassan-aws) [inline docs/src/faq.md:18]:
```suggestion
`kani::assume(false)` (or `kani::assume(cond)` where `cond` is a condition that results in `false` in the context of the program), won't cause errors in Kani.
```

**Review by zhassan-aws — APPROVED** [review]
Looks great! Thanks @adpaco-aws. Do you have a link that shows how it's rendered?

**Comment by celinval** [comment]
> Looks great! Thanks @adpaco-aws. Do you have a link that shows how it's rendered?

https://github.com/adpaco-aws/rmc/blob/docs-faq-section/docs/src/faq.md

**Review by celinval — COMMENTED** [review]
Thanks for starting this Adrian!

File: docs/src/faq.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# FAQs
+
+This section collects frequently asked questions about Kani.
+Please consider [opening an issue](https://github.com/model-checking/kani/issues/new/choose) if you have a question that would like to see here.
+
+## Questions
+
+**Question:** Kani doesn't fail after `kani::assume(false)`. Why? [Answer](#assuming-a-false-condition)
```
[the remark this answers is not in the document]
Comment (adpaco-aws) [reply docs/src/faq.md:8]: Yes, this is the other option I was talking about. I'm going to write it this way and see if it doesn't limit us in any way (don't know how it will show code snippets).

Thanks for linking to the rendered version earlier!

File: docs/src/faq.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+Instead, such an assumption has the effect of blocking all the symbolic execution paths from the assumption.
+Therefore, all checks after the assumption should appear as [`UNREACHABLE`](#../../verification-results.md).
+That's the expected behavior for `kani::assume(false)` in Kani.
+
+If you didn't expect certain checks in a harness to be `UNREACHABLE`, we recommend using the [`kani::cover` macro](#../../verification-results.md#cover-property-results) to determine what conditions are possible in case you've over-constrained the harness.
+
+### `Arbitrary` implementations for foreign types
+
+This error is due to a violation of Rust's orphan rules for trait implementations, which are explained [here](https://doc.rust-lang.org/error_codes/E0117.html).
+In that case, you'll need to follow the third approach mentioned [here](https://model-checking.github.io/kani/tutorial-nondeterministic-variables.html#custom-nondeterministic-types) to implement `Arbitrary` for a foreign custom type.
```
[the remark this answers is not in the document]
Comment (adpaco-aws) [reply docs/src/faq.md]: Added the example here with minimum details. Please take a look.

Also, another problem I had it's that the "anchored" examples didn't show up well on GitHub pages. To be clear, I'm talking about using this:
````rust
```rust
{{#include tutorial/arbitrary-variables/src/rating.rs:rating_arbitrary}}
```
````
Instead of copy-pasting the example. But I don't think that was supposed to work anyway, since we aren't actually running `mdbook`. I could use the code shown above and have a follow-up PR if it doesn't show.

[the remark this answers is not in the document]
Comment (adpaco-aws) [reply docs/src/faq.md]: Actually... the examples are not the same. The "anchored" version of `Rating` has `#[cfg_attr(kani, derive(kani::Arbitrary))]`, and I think we want to avoid that here.

File: docs/src/faq.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (adpaco-aws) [reply docs/src/faq.md]: Using HTML tags introduces some problems, but there are workarounds for most of them ([this gist](https://gist.github.com/scmx/eca72d44afee0113ceb0349dd54a84a2) explains how). For example, you cannot use back-ticks on the `summary` content. But you can use `<code>` tags.

[the remark this answers is not in the document]
Comment (celinval) [reply docs/src/faq.md]: Interesting... so you basically go into html mode at that point.

**Review by celinval — APPROVED** [review]
Looks good. Thanks
